transactor/src/main/scala/io/mediachain/transactor/JournalServer.scala

```diff
@@ -4,26 +4,33 @@ import java.util.Properties
 import java.io.FileInputStream
 import com.amazonaws.auth.BasicAWSCredentials
 import io.atomix.catalyst.transport.Address
+import io.atomix.copycat.server.CopycatServer
 import io.mediachain.copycat.{Server, Transport}
 import io.mediachain.datastore.{PersistentDatastore, DynamoDatastore}
+import scala.io.StdIn
 import scala.collection.JavaConversions._
 import sys.process._
 
 object JournalServer {
   def main(args: Array[String]) {
-    if (args.length < 1) {
-      println("Arguments: server-config-file [cluster-address ...]")
-      System.exit(1)
-    }
-    
-    val propfile = args.head
-    val cluster = args.tail.toList
+    val (interactive, config, cluster) = parseArgs(args)
     val props = new Properties
-    props.load(new FileInputStream(propfile))
-    run(props, cluster)
+    props.load(new FileInputStream(config))
+    run(interactive, props, cluster)
   }
   
-  def run(conf: Properties, cluster: List[String]) {
+  def parseArgs(args: Array[String]) = {
+    args match {
```

Comment:
given that you're `toList`ing inside anyway, why not

```scala
args.toList match {
  "-i" :: config :: cluster => ...
```

def more idiomatic scala than binding inside array extractors